packages/graphql-yoga/__tests__/context.spec.ts

```diff
@@ -137,4 +137,62 @@ describe('Context', () => {
     expect(onContextBuildingFn.mock.lastCall?.[0].context.params).toEqual(params);
     expect(onContextBuildingFn.mock.lastCall?.[0].context.request).toBeDefined();
   });
+
+  it('share the same context object', async () => {
+    const contextObjects = new Set();
+    const plugin = {
+      onContextBuilding: jest.fn(({ context }) => {
+        contextObjects.add(context);
+      }),
+      onEnveloped: jest.fn(({ context }) => {
+        contextObjects.add(context);
+      }),
+      onParse: jest.fn(({ context }) => {
+        contextObjects.add(context);
+      }),
+      onValidate: jest.fn(({ context }) => {
+        contextObjects.add(context);
+      }),
+      onExecute: jest.fn(({ args }) => {
+        contextObjects.add(args.contextValue);
+      }),
+      onRequest: jest.fn(({ serverContext }) => {
+        contextObjects.add(serverContext);
+      }),
+      onRequestParse: jest.fn(({ serverContext }) => {
+        contextObjects.add(serverContext);
+      }),
+      onResponse: jest.fn(({ serverContext }) => {
+        contextObjects.add(serverContext);
+      }),
+    };
+    const yoga = createYoga({
+      schema: createSchema({
+        typeDefs: /* GraphQL */ `
+          type Query {
+            hello: String!
+          }
+        `,
+        resolvers: {
+          Query: {
+            hello: () => 'world',
+          },
+        },
+      }),
+      plugins: [plugin],
+    });
+    const queryRes = await yoga.fetch('http://yoga/graphql?query={hello}', {
+      myExtraContext: 'myExtraContext',
+    });
+    expect(queryRes.status).toBe(200);
+    const queryResult = await queryRes.json();
+    expect(queryResult.data.hello).toBe('world');
+    expect(contextObjects.size).toBe(1);
+    for (const hook of Object.keys(plugin) as (keyof typeof plugin)[]) {
+      expect(plugin[hook]).toHaveBeenCalledTimes(1);
+    }
+    const contextObject = contextObjects.values().next().value;
+    expect(contextObject).toBeDefined();
+    expect(contextObject.myExtraContext).toBe('myExtraContext');
```

Comment:
You doesn't check that the set contains only one entry ?

Reply:
Oh I accidentially removed it 🤦

Reply:
Oh it is here;
https://github.com/dotansimha/graphql-yoga/pull/3051/commits/6e93bb1b91c5b842c66a8913e60cda5e6e801e4f#diff-c590347089564519b73f9b6ef678ef08cd1cb7d14fdfc0a4de4c231fed8785c4R190